main.tf

```diff
@@ -29,20 +29,28 @@ resource "kubernetes_deployment" "this" {
 
           resources {
             requests {
-              cpu    = var.resources.requests.cpu
-              memory = var.resources.requests.memory
+              cpu    = var.cpu_request
+              memory = var.mem_request
             }
 
             limits {
-              cpu    = var.resources.limits.cpu
-              memory = var.resources.limits.memory
+              cpu    = var.cpu_limit
+              memory = var.mem_limit
             }
           }
 
           port {
             container_port = var.port
           }
 
+          lifecycle {
+            pre_stop {
+              exec {
+                command = [ "sleep", var.pre_stop_sleep_seconds ]
```

Comment:
I think we need to also increase `terminationGracePeriodSeconds` by the amount of `pre_stop_sleep_seconds` because:
> PreStop hooks are not executed asynchronously from the signal to stop the Container; the hook must complete its execution before the TERM signal can be sent.

If we don't increase `terminationGracePeriodSeconds` then what will happen is:
```
Given terminationGracePeriodSeconds: 30
Given pre_stop_sleep_seconds: 30
Given container takes 1 second to stop after SIGTERM

[00:00] preStop hook sent: sleep 30
[00:00-30] wait for preStop to complete
[00:30] terminationGracePeriodSeconds has elapsed: forcibly kill the container
// SIGTERM never sent to container
```

In the above scenario the container never gets sent a `SIGTERM` signal as it only gets sent after the blocking `preStop` hook completes. Due to this the container is never aware that it's being asked to stop and thus doesn't run any shutdown/cleanup processes.

The scenario we want is:
```
Given terminationGracePeriodSeconds: 30
Given pre_stop_sleep_seconds: 3
Given container takes 1 second to stop after SIGTERM

[00:00] preStop hook sent: sleep 30
[00:00-29] wait for preStop to complete
[00:30] preStop completes: send SIGTERM to container
[00:31] container receives SIGTERM signal and stops after 1 second

// No need to forcibly kill the container
```